test/controllers/api/v1/fact_values_controller_test.rb

```diff
@@ -18,15 +18,15 @@ def setup
     get :index, {:host_id => @host.name }
     assert_response :success
     fact_values   = ActiveSupport::JSON.decode(@response.body)
-    expected_hash = FactValue.build_facts_hash(FactValue.where(:host_id => @host.id))
+    expected_hash = FactValue.build_facts_hash(FactValue.where(:host_id => @host.id), false)
```

Comment:
On second thought, I think these tests should not rely on calling `FactValue.build_facts_hash` to get the expected value. Could you please change them to an explicit check?
If `FactValue.build_facts_hash` returns wrong data, the tests will still pass.

Reply:
thanks, 
shouldn't the explicit check will be to build the hash exactly the same as in build_facts_hash?

Reply:
I would not build or generate the hash at all, but provide it completely hardcoded.